new-website/src/pages/partner/locations/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from "react"
+import queryString from 'query-string'
+import { EditPage } from '../../../components/location/edit-page'
+import MuiAlert from "@material-ui/lab/Alert"
+import { useIntl } from "gatsby-plugin-intl"
+import { Protected } from "../../../components/protected"
+import Layout from "../../../components/layout"
+import { Snackbar } from "@material-ui/core"
+import { ErrorBoundary } from "../../../components/error/error-boundary"
+
+function Alert({ open, severity, message, onClose }) {
+    return (
+        <Snackbar
+            anchorOrigin={{ vertical: "bottom", horizontal: "center" }}
+            autoHideDuration={6000}
+            open={open}
+            onClose={onClose}
+        >
+            <MuiAlert elevation={6} variant="filled" severity={severity}>
+                {message}
+            </MuiAlert>
+        </Snackbar>
+    )
+}
+
+const EditLocations = ({ location }) => {
+    const [alert, setAlert] = useState({ message: null, severity: "error", open: false })
+    const closeAlert = () => setAlert({ ...alert, open: false })
+
+    const { formatMessage } = useIntl()
+    const onError = ({ code }) => {
+        setAlert({
+            open: true,
+            message: formatMessage({ id: code }),
+            severity: "error",
+        })
+    }
+
+    const editParam = queryString.parse(location?.search)?.edit || 'new'
```

Comment:
What are all the possible values? 
Why `new`? not just null or undefined. This would include the case if `www.foo.de`, `www.foo.de?edit`, `www.foo.de?edit=`...

Reply:
Q&D 😄 Yes, sure why not

Reply:
On the edit page `new` is the indicator to create a new entry. It is the fallback value for `editParam` here and is used when there is no edit parameter. Rather than validating its contents here (Does the given ID exist? Is it editable for the given user?) I'd do it inside the edit page component, because there we load the document anyway.